### PR TITLE
feat: introduce proper commit linting

### LINKED ISSUE #172: Add commit lint for commit history readability and cleanliness
To enforce a clean and readable history of commit messages, commit linting should be enforced via hooks. We can look to run cleanliness checks at the CI level in the future.

- [x] Install `commitlint` with `husky` to enforce standard commit message quality
- [x] Commits that don't follow the standard `commitlint` configuration should be blocked on `commit` hook

### PR DESCRIPTION
**Pull request by temideewan**
Closes #172

### CODE REVIEW

**Comment by JamieSlome** [comment]
@temideewan - make sure to review and sign the EasyCLA 👍 This is a requirement before we can merge your PR...

File: commitlint.config.js
Comment (JamieSlome) [inline commitlint.config.js:1]: Does this not serve the same purpose as the `.commitlintrc.json` file?

Comment (temideewan) [reply commitlint.config.js:1]: Hi @JamieSlome I think they do.
I want to drop one of them instead.
Which do you think should stay on?

Comment (JamieSlome) [reply commitlint.config.js:1]: Feel free to drop `.commitlintrc.json` 👍

Comment (temideewan) [reply commitlint.config.js:1]: Done

**Comment by JamieSlome** [comment]
@temideewan, can you please merge in the latest remote changes from this repository, so we can get the vulnerability scan to run before we merge?

**Comment by temideewan** [comment]
Hi @JamieSlome 
I don't think anything has changed since when I created a fork of this repository

**Comment by JamieSlome** [comment]
@TheJuanAndOnly99, is there any reason the CVE checks aren't running on this PR?

**Comment by TheJuanAndOnly99** [comment]
@JamieSlome Strange. Looking into it. Will get back to you as soon as I find something.

**Comment by TheJuanAndOnly99** [comment]
@JamieSlome @temideewan The action is set to trigger only on "push". I've just [sent a PR](https://github.com/finos/git-proxy/pull/174) that would add the the trigger on "pull_requests" as well. I'm not sure if it will make it run on this PR however. If it doesn't we may need to close it and re send it.

**Comment by JamieSlome** [comment]
@temideewan, are you able to close and re-open the PR?

**Comment by TheJuanAndOnly99** [comment]
@JamieSlome @temideewan I don't think we need to close. I re ran the actions and the [CVE scan is working + passing](https://github.com/finos/git-proxy/actions/runs/5415211845/jobs/9844344781?pr=173)!

**Comment by temideewan** [comment]
Okay so it can be merged yeah?

I'm really excited contributing to this.

**Review by JamieSlome — APPROVED** [review]
LGTM! 🍰

**Comment by JamieSlome** [comment]
@temideewan - for some reason, the checks aren't running for me. Do you get a successful linting when you try to commit?

**Comment by temideewan** [comment]
Yes I do.
That's weird.
I'm following up on some threads about it here on GitHub though

**Comment by temideewan** [comment]
Hi @JamieSlome 
Since I can't replicate the issue on my end can you try any of the suggestions from this [stack overflow thread](https://stackoverflow.com/questions/50048717/lint-staged-not-running-on-precommit)?

**Pull request merged**